### PR TITLE
Experimental: migrate cloudbuild to buildkite

### PR DESCRIPTION
**Pull request by pcj**
Remove travis and cloudbuild, update buildkite.

I started this PR intending to replace cloudbuild with github actions.  However, as I got into it I realized there was already some overlap/duplication between the cloudbuild coverage and buildkite coverage.  Given we want to migrate everything to BK eventually anyway, I skipped the intermediate step.

> NOTE: the cloudbuild account currently in use for rules_docker also hosts google internal jobs and logs cannot be shared.  As a result, they are not suitable for community usage.

### CODE REVIEW

**Comment by pcj** [comment]
@googlebot I fixed it.

**Comment by pcj** [comment]
@googlebot I fixed it.

**Comment by pcj** [comment]
Note: turning on `--remote` in github action caused:

```
ERROR: Failed to init auth credentials: The Application Default Credentials are not available. They are available if running in Google Compute Engine. Otherwise, the environment variable GOOGLE_APPLICATION_CREDENTIALS must be defined pointing to a file defining the credentials. See https://developers.google.com/accounts/docs/application-default-credentials for more information.
```

Need to resolve this (add a secret) to use remote execution in this context.

**Comment by pcj** [comment]
List of failing tests on github action:

`//tests/docker/package_managers:install_pkgs_reproducibility_test`

> Does not necessarily imply the test is bad, just that it is not passing in github action

Planning to exclude these

File: .bazelci/presubmit.yml [hunk 1]
```diff
@@ -10,248 +10,252 @@ buildifier:
 platforms:
```
Comment (gravypod) [inline .bazelci/presubmit.yml:10]: It seems like this file was reformatted which makes it a little hard to find diffs in. Couild you tag any changes you've made?

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+          key: bazel-tests_slash_container
+      - name: Install bazelisk
+        run: |
+          curl -LO "https://github.com/bazelbuild/bazelisk/releases/download/v1.7.5/bazelisk-linux-amd64"
+          mkdir -p "${GITHUB_WORKSPACE}/bin/"
+          mv bazelisk-linux-amd64 "${GITHUB_WORKSPACE}/bin/bazel"
+          chmod +x "${GITHUB_WORKSPACE}/bin/bazel"
+      - name: Test
+        run: |
+          "${GITHUB_WORKSPACE}/bin/bazel" test \
```
Comment (gravypod) [inline .github/workflows/ci.yaml:68]: Would it make sense to group some of this into a non-yamly thing? I've had very good experiences with [test_suite](https://docs.bazel.build/versions/master/be/general.html#test_suite) in the past. If it's annoying to do in this PR we can leave it as is until it becomes annoying to keep in sync.

**Review by gravypod — COMMENTED** [review]
Any idea where that credential/token that you need might come from? If not we might need to ping the previous maintainers.

File: .github/workflows/ci.yaml (hunk 2 above)
Comment (pcj) [reply .github/workflows/ci.yaml:74]: test_suite is a good idea!

File: .bazelci/presubmit.yml (hunk 1 above)
Comment (pcj) [reply .bazelci/presubmit.yml:10]: Yes I'll fix the formatting ultimately.  PR is still in a draft state (probably should have made it a draft PR)

File: .github/workflows/ci.yaml (hunk 2 above)
Comment (pcj) [reply .github/workflows/ci.yaml:74]: I'm also contemplating a tool to generate the YAML here.  Looking at the presubmit and trying to see what's overlapping.

**Comment by pcj** [comment]
Full list of tests:

```
$ bazel query 'kind(".*_test", //...)' --output=label_kind
sh_test rule //tests/docker/util:test_extracted_file
_file_test_rule rule //tests/docker/util:test_extracted_file_impl
sh_test rule //tests/docker/util:test_extract_script
_file_test_rule rule //tests/docker/util:test_extract_script_impl
sh_test rule //tests/docker/util:test_container_extract_rule
_rule_test_rule rule //tests/docker/util:test_container_extract_rule_impl
sh_test rule //tests/docker/util:test_container_commit_rule
_rule_test_rule rule //tests/docker/util:test_container_commit_rule_impl
_container_test rule //tests/docker/util:test_container_commit_metadata
sh_test rule //tests/docker/util:test_container_commit_layer_rule
_rule_test_rule rule //tests/docker/util:test_container_commit_layer_rule_impl
_container_test rule //tests/docker/util:test_container_commit_layer_metadata
_container_test rule //tests/docker/toolchain_container:test-rbe-test-xenial-with-pkgs
sh_test rule //tests/docker/security:security_check_test
_file_test_rule rule //tests/docker/security:security_check_test_impl
sh_test rule //tests/docker/security:security_check_expect_cve_test
_file_test_rule rule //tests/docker/security:security_check_expect_cve_test_impl
sh_test rule //tests/docker/package_managers:test_install_pkgs_rule
_rule_test_rule rule //tests/docker/package_managers:test_install_pkgs_rule_impl
sh_test rule //tests/docker/package_managers:test_download_pkgs_rule
_rule_test_rule rule //tests/docker/package_managers:test_download_pkgs_rule_impl
sh_test rule //tests/docker/package_managers:test_download_pkgs_docker_run
_file_test_rule rule //tests/docker/package_managers:test_download_pkgs_docker_run_impl
sh_test rule //tests/docker/package_managers:test_download_pkgs_docker_cp
_file_test_rule rule //tests/docker/package_managers:test_download_pkgs_docker_cp_impl
compare_ids_test rule //tests/docker/package_managers:install_pkgs_reproducibility_test
_container_test rule //tests/docker/package_managers:gpg_test
sh_test rule //tests/docker/package_managers:download_pkgs_with_additional_repos_run_test
sh_test rule //tests/docker/package_managers:download_pkgs_run_test
sh_test rule //tests/contrib/automatic_container_release:test_packages_metadata
_file_test_rule rule //tests/contrib/automatic_container_release:test_packages_metadata_impl
sh_test rule //tests/contrib/automatic_container_release:test_metadata_merge
_file_test_rule rule //tests/contrib/automatic_container_release:test_metadata_merge_impl
configs_test rule //tests/contrib/automatic_container_release:configs_test_deps_spec_only
configs_test rule //tests/contrib/automatic_container_release:configs_test
compare_ids_test rule //tests/contrib:test_id_compare_ids_test
_container_test rule //tests/contrib:test_dockerfile_image_docker
compare_ids_fail_test rule //tests/contrib:test_compare_ids_test_wrong_id_fails
compare_ids_fail_test rule //tests/contrib:test_compare_ids_test_one_tar_no_id_fails
compare_ids_fail_test rule //tests/contrib:test_compare_ids_test_no_images_with_id_fails
compare_ids_fail_test rule //tests/contrib:test_compare_ids_test_invalid_tar_fails_multi_regex
compare_ids_fail_test rule //tests/contrib:test_compare_ids_test_invalid_tar_fails
compare_ids_fail_test rule //tests/contrib:test_compare_ids_test_diff_ids_fails_no_regex
compare_ids_fail_test rule //tests/contrib:test_compare_ids_test_diff_ids_fails
compare_ids_test rule //tests/contrib:test_compare_ids_test
container_repro_test rule //tests/contrib:set_cmd_repro_test
sh_test rule //tests/contrib:rename_image_test
container_repro_test rule //tests/contrib:rbe-test-xenial_repro_test
container_repro_test rule //tests/contrib:random_file_img_non_repro_test
container_repro_test rule //tests/contrib:derivative_with_volume_repro_test
_container_test rule //tests/container/scala:scala_image_test
_container_test rule //tests/container/rust:rust_image_test
_container_test rule //tests/container/python3:py3_image_test
_container_test rule //tests/container/python:py_image_test
_container_test rule //tests/container/nodejs:nodejs_image_test
_container_test rule //tests/container/nodejs:nodejs_image_none_args_test
_container_test rule //tests/container/nodejs:nodejs_image_no_args_test
_container_test rule //tests/container/nodejs:nodejs_image_list_with_empty_string_args_test
_container_test rule //tests/container/nodejs:nodejs_image_empty_list_args_test
_container_test rule //tests/container/nodejs:nodejs_image_custom_binary_with_args_test
_container_test rule //tests/container/nodejs:nodejs_image_custom_binary_test
_container_test rule //tests/container/kotlin:kotlin_image_test
_container_test rule //tests/container/java:simple_java_entrypoint_image_test
_container_test rule //tests/container/java:java_runfiles_image_test
_container_test rule //tests/container/java:java_runfiles_as_lib_image_test
_container_test rule //tests/container/java:java_partial_entrypoint_image_test
_container_test rule //tests/container/java:java_image_test
_container_test rule //tests/container/groovy:groovy_image_test
_container_test rule //tests/container/go:go_static_image_test
_container_test rule //tests/container/go:go_image_test
_container_test rule //tests/container/cc:cc_image_test
sh_test rule //tests/container:test_push_digest_output
_file_test_rule rule //tests/container:test_push_digest_output_impl
sh_test rule //tests/container:test_digest_output2
_file_test_rule rule //tests/container:test_digest_output2_impl
sh_test rule //tests/container:test_digest_output1
_file_test_rule rule //tests/container:test_digest_output1_impl
_container_test rule //tests/container:structure_test
_container_test rule //tests/container:stripped_directory_name_test
_container_test rule //tests/container:special_characters_test
_container_test rule //tests/container:set_env_make_vars_test
_container_test rule //tests/container:set_env_equals_test
_container_test rule //tests/container:set_env_csv_test
_container_test rule //tests/container:set_entrypoint_test
_container_test rule //tests/container:set_cmd_test_host
_container_test rule //tests/container:set_cmd_test
_container_test rule //tests/container:set_cmd_and_entrypoint_test
_container_test rule //tests/container:set_cmd_and_entrypoint_null_on_base_test
pull_info_validation_test rule //tests/container:pull_info_validation_test
sh_test rule //tests/container:pause_tar_test_config
_file_test_rule rule //tests/container:pause_tar_test_config_impl
sh_test rule //tests/container:pause_tar_test_1_tar_gz
_file_test_rule rule //tests/container:pause_tar_test_1_tar_gz_impl
sh_test rule //tests/container:pause_tar_test_0_tar_gz
_file_test_rule rule //tests/container:pause_tar_test_0_tar_gz_impl
_container_test rule //tests/container:null_cmd_and_entrypoint_none_test
_container_test rule //tests/container:null_cmd_and_entrypoint_empty_str_test
_container_test rule //tests/container:null_cmd_and_entrypoint_empty_list_test
sh_test rule //tests/container:new_distroless_base_oci_test_config
_file_test_rule rule //tests/container:new_distroless_base_oci_test_config_impl
sh_test rule //tests/container:new_distroless_base_oci_test_0_tar_gz
_file_test_rule rule //tests/container:new_distroless_base_oci_test_0_tar_gz_impl
_container_test rule //tests/container:new_alpine_linux_ppc64le_image_tar_test
_container_test rule //tests/container:new_alpine_linux_ppc64le_image_oci_test
_container_test rule //tests/container:new_alpine_linux_ppc64le_image_oci_tar_test
_container_test rule //tests/container:new_alpine_linux_ppc64le_image_oci_go_join_layers_test
_container_test rule //tests/container:new_alpine_linux_armv6_image_tar_test
sh_test rule //tests/container:new_alpine_linux_armv6_image_tar_digest_test
_file_test_rule rule //tests/container:new_alpine_linux_armv6_image_tar_digest_test_impl
_container_test rule //tests/container:new_alpine_linux_armv6_image_oci_test
_container_test rule //tests/container:new_alpine_linux_armv6_image_oci_tar_test
sh_test rule //tests/container:k8s_pause_arm64_digest_test
_file_test_rule rule //tests/container:k8s_pause_arm64_digest_test_impl
sh_test rule //tests/container:distroless_fixed_id_image_digest_test
_file_test_rule rule //tests/container:distroless_fixed_id_image_digest_test_impl
sh_test rule //tests/container:distroless_fixed_id_digest_test
_file_test_rule rule //tests/container:distroless_fixed_id_digest_test_impl
_container_test rule //tests/container:deb_image_with_dpkgs_test
py_test rule //tests/container:build_tar_test
_container_test rule //tests/container:basic_windows_image_test
_container_test rule //tests/container:base_with_cmd_and_entrypoint_test
sh_test rule //tests/container:architecture_test
_container_test rule //tests/container:alpine_ppc64le_image_test
sh_test rule //tests/container:alpine_linux_armv6_tar_test_image_tar
_file_test_rule rule //tests/container:alpine_linux_armv6_tar_test_image_tar_impl
_container_test rule //tests/container:alpine_linux_amd64_image_tar_test
_container_test rule //tests/container:alpine_linux_amd64_image_oci_test
_container_test rule //tests/container:alpine_linux_amd64_image_oci_tar_test
sh_test rule //tests/container:alpine_custom_attr_digest_test
_file_test_rule rule //tests/container:alpine_custom_attr_digest_test_impl
_container_test rule //tests/container:alpine_arm_image_test
_container_test rule //tests/container:alpine_amd64_image_test
_container_test rule //testing/new_pusher_tests:new_push_verify_pushed_configs_and_files
py_test rule //docker/util:config_stripper_test
go_test rule //container/go/pkg/oci:go_default_test
go_test rule //container/go/pkg/compat:go_default_test
py_test rule //container:image_test
_container_test rule //:structure_test_at_workspace_root
```

**Comment by pcj** [comment]
Failed on buildkite:

```
      - //testing/new_pusher_tests:new_push_verify_pushed_configs_and_files
```

**Comment by pcj** [comment]
Nice, iteration 8 passed
Screen Shot 2021-03-16 at 11 34 25 PM

**Comment by pcj** [comment]
Not sure what this tool is about, but it does not look like we can pull `gcr.io/asci-toolchains`. 

```
steps:
  - name: "l.gcr.io/google/bazel"
    args: 
      - "--bazelrc=.bazelrc"
      - "test"
      - "--test_output=errors"
      - "--verbose_failures"
      - "//tests/contrib:test_dockerfile_image_docker"
      - "--config=remote"
  # Pull the semantic checker image used by configs_test targets.
  - name: "gcr.io/cloud-builders/docker"
    args: ["pull", "gcr.io/asci-toolchain/container_release_tools/dependency_update/validators/semantics:latest"]
    id: "pull_validator"
  - name: "l.gcr.io/google/bazel"
    args: 
      - "test"
      - "--test_output=errors"
      - "--verbose_failures"
      - "//tests/contrib/automatic_container_release:all"
```

> NOTE: //tests/contrib:test_dockerfile_image_docker is already running in buildkite

**Comment by pcj** [comment]
Issue migrating `//run_instruction_arbitrary:all` to buildkite:

```
(06:25:53) ERROR: /workdir/testing/examples/run_instruction_arbitrary/BUILD:65:16: //run_instruction_arbitrary:bazel_gcloud_dockerfile_wrapper depends on @bazel_gcloud_dockerfile//image:dockerfile_image.tar in repository @bazel_gcloud_dockerfile which failed to fetch. no such package '@bazel_gcloud_dockerfile//image': docker build command failed: The command '/bin/sh -c wget https://bootstrap.pypa.io/3.5/get-pip.py && python3 get-pip.py && python3 -m pip install --upgrade setuptools wheel' returned a non-zero code: 1
```

**Comment by pcj** [comment]
OK, pausing work on this for now.  Here's where it stands: I migrated a bunch of stuff off cloudbuild to buildkite without too much difficulty.  The remaining ones have special requirements and/or I don't know what they are supposed to do:

1. `tests/contrib/cloudbuild.yaml`: this pulls in `gcr.io/asci-toolchain/container_release_tools/dependency_update/validators/semantics:latest` to help test `//tests/contrib/automatic_container_release:all`.  I need more context on this.
2.  `tests/docker/security/cloudbuild.yaml` requires `gcr.io/asci-toolchain/nosla-ubuntu16_04-bazel-docker-gcloud` to test `//tests/docker/security/...`.  Again, need more context here.
3. `container/go/cloudbuild.yaml` builds the puller for multiple platforms and uploadsthe artifacts.  This is fairly straightforward to run in BK, but are these artifacts used by downstream clients somewhere?  Need more info about this.
4. `testing/examples/run_instruction_apt_pkgs/cloudbuild.yaml` is somewhat more complex with a custom bazel output_base.  It looks a bit more daunting so I skipped it for the moment.
5. `testing/examples/run_instruction_apt_pkgs/cloudbuild.yaml` failed on buildkite probably because the BK image is missing more or more tools in `/bin/sh -c wget https://bootstrap.pypa.io/3.5/get-pip.py && python3 get-pip.py && python3 -m pip install --upgrade setuptools wheel' returned a non-zero code: 1`.

Also, at this intermediate stage the presubmit is only testing on ubuntu.  Need to restore the macos coverage.

**Comment by pcj** [comment]
@smukherj1 do you want to sync on this sometime later in the week?

**Comment by pcj** [comment]
cc @alexeagle

**Comment by alexeagle** [comment]
Looks like we should discuss at next maintainer meeting? Do you want a review on this now?

**Comment by pcj** [comment]
Resolving some of those unknowns I enumerated above best during a meeting.  Any other PR comments certainly welcome.

**Comment by pcj** [comment]
The ubuntu keyserver gpg key sha256 changed.  Here is the value I am currently seeing:

```
-----BEGIN PGP PUBLIC KEY BLOCK-----
Comment: Hostname: 
Version: Hockeypuck ~unreleased

xo0ES8OmlQEEALy8ttT3KgmjoqCkeU7S04/j615RDivV0CHv+5mdJFQY10wo6v6k
rmXxK757eIoRDN1B3ztl8vjqLHxi8oA+f4inZcrrIZYjW0MghO+IcKEbKCrPUPjD
fQBSIvZD5ZaiibMNwZJ6TidS2r6nSAH+aoMRgXYycQPGkAHDcaPKY54zABEBAAHN
JExhdW5jaHBhZCBPcGVuSkRLIGJ1aWxkcyAoYWxsIGFyY2hzKcK2BBMBAgAgBQJL
w6aVAhsDBgsJCAcDAgQVAggDBBYCAwECHgECF4AACgkQ65sdiIb0TiraSgQAr+WD
+qGcW5yrkS2QBMq+g+Ew2tLqeDLw8Y2xZ4BhnRzB3WljBfwj7wt/KOugjJWaEWuY
GiuLBf6vy+At5VtXs0FSk6oIZR0XZvUVamFPXeTkZqUiw4oTNlBTeOhAxrevRWzn
mLuRAIWlr0dARJyLCK5MMVVhesouoG8WpOrniXs=
=JbYw
-----END PGP PUBLIC KEY BLOCK-----
```

```python
http_file(
    name = "launchpad_openjdk_gpg",
    sha256 = "e9a596d0c194a562be9fd2c2a0994d7885505a1145fed0fbd5ae4c11d56220a0",
    urls = ["https://keyserver.ubuntu.com/pks/lookup?op=get&fingerprint=on&search=0xEB9B1D8886F44E2A"],
)
```

**Comment by pcj** [comment]
Subsumed by #1859

**Pull request closed without merging**